app/models/application_choice.rb

```diff
@@ -185,9 +178,40 @@ def withdrawn_at_candidates_request?
     )
   end
 
+  def configure_initial_course_choice!(course_option)
+    update!(
+      course_option: course_option,
+      current_course_option: course_option,
+    )
+    update_course_option!(course_option)
+  end
+
+  def update_course_option!(new_course_option, other_fields: {}, audit_comment: nil)
```

Comment:
I was wondering if we could rename this method to indicate we are also updating the providers and recruitment year, maybe something around updating course options and lookups etc.

Reply:
How about `update_course_option_and_associated_fields!`?